swift/trainers/push_to_ms.py

```diff
@@ -0,0 +1,179 @@
+
+    @staticmethod
+    @future_compatible
+    def upload_folder(
+        self,
+        *,
+        repo_id: str,
+        folder_path: Union[str, Path],
+        path_in_repo: Optional[str] = None,
+        commit_message: Optional[str] = None,
+        commit_description: Optional[str] = None,
+        token: Union[str, bool, None] = None,
+        revision: Optional[str] = 'master',
+        ignore_patterns: Optional[Union[List[str], str]] = None,
+        run_as_future: bool = False,
+        **kwargs,
+    ) -> Union[CommitInfo, str, Future[CommitInfo], Future[str]]:
+        commit_message = commit_message or 'Upload folder using api'
+        if commit_description:
+            commit_message = commit_message + '\n' + commit_description
+        if not os.path.exists(os.path.join(folder_path, 'configuration.json')):
+            with open(os.path.join(folder_path, 'configuration.json'), 'w') as f:
+                f.write('{"framework": "pytorch", "task": "text-generation", "allow_remote": true}')
+        if ignore_patterns:
+            ignore_patterns = [p for p in ignore_patterns if p != '_*']
+        if path_in_repo:
+            idx = folder_path.rfind(path_in_repo)
+            if idx >= 0:
+                folder_path = folder_path[:idx]
+            ignore_patterns = []
+        push_to_hub(
+            repo_id,
+            folder_path,
+            token or PushToMsHubMixin._token,
+            commit_message=commit_message,
+            ignore_file_pattern=ignore_patterns,
+            revision=revision,
+            tag=path_in_repo)
+        return CommitInfo(
+            commit_url=f'https://www.modelscope.cn/models/{repo_id}/files',
+            commit_message=commit_message,
+            commit_description=commit_description,
+            oid=None,
+        )
+
+    if not _use_hf_hub:
+        import huggingface_hub
+        from huggingface_hub.hf_api import api
+        from transformers import trainer
+        huggingface_hub.create_repo = create_repo
+        huggingface_hub.upload_folder = partial(upload_folder, api)
+        trainer.create_repo = create_repo
+        trainer.upload_folder = partial(upload_folder, api)
+
+    @staticmethod
+    def _create_ms_repo(hub_model_id: str, hub_token: Optional[str] = None, hub_private_repo: bool = False) -> str:
+        assert hub_model_id is not None, 'Please enter a valid hub_model_id'
+
+        api = HubApi()
+        if hub_token is None:
+            hub_token = os.environ.get('MODELSCOPE_API_TOKEN')
+        if hub_token is not None:
+            api.login(hub_token)
+        visibility = ModelVisibility.PRIVATE if hub_private_repo else ModelVisibility.PUBLIC
+
+        if '/' not in hub_model_id:
+            user_name = ModelScopeConfig.get_user_info()[0]
+            assert isinstance(user_name, str)
+            hub_model_id = f'{user_name}/{hub_model_id}'
+            logger.info(f"'/' not in hub_model_id, setting hub_model_id: {hub_model_id}")
```

Comment:
'/' not in hub_model_id, pushing to personal repo {hub_model_id}

Reply:
Fixed